docs/architecture/porting-existing-aspnet-apps/example-migration-eshop.md

```diff
@@ -0,0 +1,846 @@
+
+```csharp
+  // add using Microsoft.Extensions.Configuration
+  public CatalogDBInitializer(CatalogItemHiLoGenerator indexGenerator,
+      IConfiguration configuration)
+  {
+      this.indexGenerator = indexGenerator;
+      useCustomizationData = configuration.GetValue<bool>("UseCustomizationData");
+  }
+```
+
+All access to configuration within the web app should be modified in this manner to use the new `IConfiguration` type. Dependencies that require access to .NET Framework configuration can include such settings in an *app.config* file added to the web project. The dependent projects can work with `ConfigurationManager` to access settings, and shouldn't require any changes if they already use this approach. However, since ASP.NET Core apps run as their own executable, they don't reference *web.config* but rather *app.config*. By migrating settings from the legacy app's *web.config* file to a new *app.config* file in the ASP.NET Core app, components that use `ConfigurationManager` to access their settings will continue to function properly.
+
+The app's migration is nearly complete. The only remaining task is data access configuration.
+  
+## Data access considerations
+
+ASP.NET Core apps running on .NET Framework can continue to leverage Entity Framework (EF). If performing an incremental migration, getting the app working with EF 6 before trying to port its data access to use EF Core may be worthwhile. In this way, any problems with the app's migration can be identified and addressed before another block of migration effort is begun.
+
+As it happens, configuring EF 6 in the eShop sample migration doesn't require any special work, since this work was performed in the Autofac `ApplicationModule`. The only problem is that currently the `CatalogDBContext` class tries to read its connection string from *web.config*. To address this, the connection details need to be added to *appsettings.json*. Then the connection string must be passed into `CatalogDBContext` when it's created.
+
+First, *appsettings.json* is updated to include the connection string. The full file is listed here:
+
+```json
+{
+  "ConnectionStrings": {
+    "DefaultConnection": "Server=(localdb)\\mssqllocaldb;Database=eShopPorted;Trusted_Connection=True;MultipleActiveResultSets=true"
+  },
+  "Logging": {
+    "LogLevel": {
+      "Default": "Warning"
+    }
+  },
+  "AllowedHosts": "*",
+  "UseMockData": "false",
+  "UseCustomizationData": "true"
+}
+```
+
+Next, the connection string must be passed into the constructor when the `DbContext` is created. Since the instances are created by Autofac, the change needs to be made in `ApplicationModule`. Modify the module to take in a `connectionString` in its constructor and assign it to a field. Then modify the registration for `CatalogDBContext` to add connection string as a parameter:
```

Comment:
```suggestion
The connection string must be passed into the constructor when the `DbContext` is created. Since the instances are created by Autofac, the change needs to be made in `ApplicationModule`. Modify the module to take in a `connectionString` in its constructor and assign it to a field. Then modify the registration for `CatalogDBContext` to add connection string as a parameter:
```